assets/pages/new-contributions-landing/components/ContributionForm.jsx

```diff
@@ -148,7 +184,14 @@ function onSubmit(props: PropTypes): Event => void {
           break;
 
         case 'PayPal':
-          // TODO
+          if (props.contributionType === 'ONE_OFF') {
```

Comment:
This whole function could benefit from the same kind of cleanup you have done elsewhere by providing a matrix of payment handlers (indexed by payment method and contribution type).

Reply:
I like this suggestion a lot - I'll see about adding it as part of PayPal recurring